src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp

```diff
@@ -208,18 +303,30 @@ void ov::npuw::LLMInferRequest::infer_prefill(ov::SoPtr<ov::ITensor> input_ids,
         padded_attention_mask->data<int64_t>() + padded_attention_mask->get_size() - attention_mask->get_size());
 
     auto padded_position_ids = m_prefill_request->get_tensor(m_prefill_in_ports.at("position_ids"));
-
     std::copy_n(position_ids->data<int64_t>(),
                 position_ids->get_size(),
                 padded_position_ids->data<int64_t>() + padded_position_ids->get_size() - position_ids->get_size());
 
     m_prefill_request->infer();
+    kvcache_desc.num_stored_tokens += static_cast<uint32_t>(input_ids->get_shape()[INPUT_IDS_SEQ_LEN_DIM]);
+
+    if (m_tail_mm_request) {
+        LOG_DEBUG("Calling inference for tail model asynchronously...");
+        m_tail_mm_request->set_tensor(m_tail_embed_port,
+                                      m_prefill_request->get_tensor(m_prefill_out_ports.at("output_embed")));
+        m_tail_mm_request->start_async();
+        copy_kvcache();
+        m_tail_mm_request->wait();
+        LOG_DEBUG("Calling inference for tail model - done.");
+
+        m_logits = m_tail_mm_request->get_tensor(m_tail_logits_port);
+    } else {
+        copy_kvcache();
 
-    m_npuw_llm_compiled_model->m_kvcache_desc.num_stored_tokens +=
-        static_cast<uint32_t>(input_ids->get_shape()[INPUT_IDS_SEQ_LEN_DIM]);
-    m_need_copy_kvcache = true;
+        m_logits = m_prefill_request->get_tensor(m_prefill_out_ports.at("logits"));
+    }
 
-    m_logits = m_prefill_request->get_tensor(m_prefill_out_ports.at("logits"));
+    m_once_per_generate = std::make_unique<std::once_flag>();
```

Comment:
Not sure why do you need to complicate it like this. If the whole idea is to initialize the generate model's attention mask here, you can make it right after prefill execution (right here). It won't affect the TTFT much.